src/Home/ProgressBar.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { FunctionComponent } from "react"
+import { View, StyleSheet } from "react-native"
+
+import * as Habit from "../habit"
+import * as ThemeColor from "../themeColor"
+import { TOTAL_NUMBER_OF_DAYS, OUTER_PADDING } from "./constants"
+
+import { Sizing, Colors, Outlines } from "../styles"
+
+interface ProgressBarProps {
+  habit: Habit.Habit
+}
+
+const ProgressBar: FunctionComponent<ProgressBarProps> = ({ habit }) => {
+  const daysCompleted =
+    habit.history.length > TOTAL_NUMBER_OF_DAYS
+      ? TOTAL_NUMBER_OF_DAYS
+      : habit.history.length
+
+  const totalProgressBarWidth = Sizing.screen.width - 4 * OUTER_PADDING
```

Comment:
I don't think this is part of the refactor technically but we may want to name `4 * OUTER_PADDING`. Maybe something like:

```suggestion
  const spaceOutsideProgressBar = 4 * OUTER_PADDING
  const totalProgressBarWidth = Sizing.screen.width - spaceOutsideProgressBar
```